decidim/decidim
# Change the custom public port ENV variable name to HTTP_PORT

#### :tophat: What? Why?
As described at #9559, the asset URLs are pointing to the correct host with an incorrect port name. Actually this concerns all URLs generated using the `EngineRouter` as described at #9558.

This happens when Puma is running separately in a separate port which is forwarded to the public internet using an HTTP server in front which forwards all requests to a specific domain to this port on the local machine.

This causes e.g. Heroku to produce incorrect asset URLs or any other URLs created using `EngineRouter`.

This fixes the issue by renaming the `PORT` ENV variable (introduced in #9403) to `HTTP_PORT` which is not defined at Heroku.

#### :pushpin: Related Issues
- Fixes #9559
- Fixes #9558

#### Testing
- Run the Decidim rails server (i.e. Puma) in a custom port, not 80 or 443
- Run the Decidim server in a non-development and non-test environment using e.g. `RAILS_ENV=production` (and configure it accordingly)
- Add a HTTP server (such as nginx) in front of Decidim and forward the requests coming to port `:80` ([see configuration example for nginx](https://github.com/puma/puma/blob/master/docs/nginx.md))

##### For testing the favicon/asset URLs
- Request Decidim using port `:80`
- Upload a favicon to Decidim from the admin panel
- See that the issue described at #9559 no longer exists

##### For testing the other URLs generated through EngineRouter
- Using the same configuration as before
- Start the Rails console using `PORT=1234 RAILS_ENV=production bundle exec rails c` (make sure spring is restarted if you're using it, run `spring stop` before starting the console)
- Run `Decidim::ResourceLocatorPresenter.new(Decidim::Proposals::Proposal.first).url`
- See that the URL is generated correctly without the port `1234` and the issue described at #9558 no longer exists

Or alternatively, find one of these routes through the UI, these should be all over, e.g. in notifications, emails, exports, etc.
